AudioSinkExample/ViewController.swift

```diff
@@ -0,0 +1,692 @@
+            let recognizerDoubleTap = UITapGestureRecognizer(target: self, action: #selector(ViewController.changeRemoteVideoAspect))
+            recognizerDoubleTap.numberOfTapsRequired = 2
+            remoteView.addGestureRecognizer(recognizerDoubleTap)
+
+            // Single tap to recognize remote audio.
+            let recognizerTap = UITapGestureRecognizer(target: self, action: #selector(ViewController.recognizeRemoteAudio))
+            recognizerTap.require(toFail: recognizerDoubleTap)
+            remoteView.addGestureRecognizer(recognizerTap)
+
+            // Start rendering, and add to our stack.
+            publication.remoteTrack?.addRenderer(remoteView)
+            self.remoteViewStack.addArrangedSubview(remoteView)
+        }
+    }
+
+    func removeRemoteVideoView(publication: TVIRemoteVideoTrackPublication) {
+        let viewTag = publication.trackSid.hashValue
+        if let remoteView = self.remoteViewStack.viewWithTag(viewTag) {
+            // Stop rendering, we don't want to receive any more frames.
+            publication.remoteTrack?.removeRenderer(remoteView as! TVIVideoRenderer)
+            // Automatically removes us from the UIStackView's arranged subviews.
+            remoteView.removeFromSuperview()
+        }
+    }
+
+    func changeRemoteVideoAspect(gestureRecognizer: UIGestureRecognizer) {
+        guard let remoteView = gestureRecognizer.view else {
+            print("Couldn't find a view attached to the tap recognizer. \(gestureRecognizer)")
+            return;
+        }
+
+        if (remoteView.contentMode == .scaleAspectFit) {
+            remoteView.contentMode = .scaleAspectFill
+        } else {
+            remoteView.contentMode = .scaleAspectFit
+        }
+    }
+}
+
+// MARK: UITextFieldDelegate
+extension ViewController : UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        self.connect(sender: textField)
+        return true
+    }
+}
+
+// MARK: TVIRoomDelegate
+extension ViewController : TVIRoomDelegate {
+    func didConnect(to room: TVIRoom) {
+
+        // Listen to events from existing `TVIRemoteParticipant`s
+        for remoteParticipant in room.remoteParticipants {
+            remoteParticipant.delegate = self
+        }
+
+        // Wait until our LocalAudioTrack is assigned a SID to record it.
+        if (recordAudio) {
+            if let localParticipant = room.localParticipant {
+                localParticipant.delegate = self
+            }
+
+            if let localAudioPublication = room.localParticipant?.localAudioTracks.first,
+               let localAudioTrack = localAudioPublication.localTrack {
+                let trackSid = localAudioPublication.trackSid
+                self.audioRecorders[trackSid] = ExampleAudioRecorder.init(audioTrack: localAudioTrack,
+                                                                          identifier: trackSid)
+            }
```

Comment:
move code from line 495 to 500 inside 492 check - room.localParticipant does not need to unwrap then

Reply:
I'm not sure if this is an improvement, since setting the delegate is a somewhat separate operation.